internal/services/mobilenetwork/mobile_network_packet_core_control_plane_resource.go

```diff
@@ -0,0 +1,687 @@
+	output := packetcorecontrolplane.LocalDiagnosticsAccessConfiguration{
+		AuthenticationType: packetcorecontrolplane.AuthenticationType(model.AuthenticationType),
+	}
+	if model.HttpsServerCertificateUrl != "" {
+		output.HTTPSServerCertificate = &packetcorecontrolplane.HTTPSServerCertificate{
+			CertificateUrl: model.HttpsServerCertificateUrl,
+		}
+	}
+	return output
+}
+
+func flattenLocalPacketCoreControlLocalDiagnosticsAccessConfiguration(input packetcorecontrolplane.LocalDiagnosticsAccessConfiguration) []LocalDiagnosticsAccessConfigurationModel {
+	outputs := make([]LocalDiagnosticsAccessConfigurationModel, 0)
+	output := LocalDiagnosticsAccessConfigurationModel{
+		AuthenticationType: string(input.AuthenticationType),
+	}
+	if input.HTTPSServerCertificate != nil {
+		output.HttpsServerCertificateUrl = input.HTTPSServerCertificate.CertificateUrl
+	}
+	outputs = append(outputs, output)
+	return outputs
+}
+
+func expandPacketCoreControlPlaneInterfacePropertiesModel(inputList []InterfacePropertiesModel) packetcorecontrolplane.InterfaceProperties {
+	output := packetcorecontrolplane.InterfaceProperties{}
+	if len(inputList) == 0 {
+		return output
+	}
+
+	input := inputList[0]
+
+	if input.IPv4Address != "" {
+		output.IPv4Address = &input.IPv4Address
+	}
+
+	if input.IPv4Gateway != "" {
+		output.IPv4Gateway = &input.IPv4Gateway
+	}
+
+	if input.IPv4Subnet != "" {
+		output.IPv4Subnet = &input.IPv4Subnet
+	}
+
+	if input.Name != "" {
+		output.Name = &input.Name
+	}
+
+	return output
+}
+
+func expandPlatformConfigurationModel(inputList []PlatformConfigurationModel) (packetcorecontrolplane.PlatformConfiguration, error) {
+	output := packetcorecontrolplane.PlatformConfiguration{}
+	if len(inputList) == 0 {
+		return output, nil
+	}
+
+	input := inputList[0]
+
+	output.Type = packetcorecontrolplane.PlatformType(input.Type)
+
+	if pass, err := vertifyPlatformConfigurationModel(input); !pass {
+		return output, err
+	}
+
+	if input.AzureStackEdgeDeviceId != "" {
+		output.AzureStackEdgeDevice = &packetcorecontrolplane.AzureStackEdgeDeviceResourceId{
+			Id: input.AzureStackEdgeDeviceId,
+		}
+	}
+
+	if input.ConnectedClusterId != "" {
+		output.ConnectedCluster = &packetcorecontrolplane.ConnectedClusterResourceId{
+			Id: input.ConnectedClusterId,
+		}
+	}
+
+	if input.AzureStackHciClusterId != "" {
+		output.AzureStackHciCluster = &packetcorecontrolplane.AzureStackHCIClusterResourceId{
+			Id: input.AzureStackHciClusterId,
+		}
+	}
+
+	if input.CustomLocationId != "" {
+		output.CustomLocation = &packetcorecontrolplane.CustomLocationResourceId{
+			Id: input.CustomLocationId,
+		}
+	}
+
+	return output, nil
+}
+
+func vertifyPlatformConfigurationModel(input PlatformConfigurationModel) (bool, error) {
+	idList := make([]string, 0)
+	if input.AzureStackEdgeDeviceId != "" {
+		idList = append(idList, input.AzureStackEdgeDeviceId)
+	}
+	if input.AzureStackHciClusterId != "" {
+		idList = append(idList, input.AzureStackHciClusterId)
```

Comment:
(as above) does the Resource ID come back incorrectly from the Azure API? does this need rewriting?